app/views/product_users/_users_results.html.haml

```diff
@@ -0,0 +1,14 @@
+- if @product_users.nil?
+- elsif @product_users.empty?
+  -if @search_action_performed
+    %p.notice= text("no_results")
+  - else
+    %p.notice= text("empty_access_list")
```

Comment:
Then I think we need this logic to cover all cases. maybe with some code comments:
```suggestion
- if @product_users.nil?
  - # Access list not required for this product
  %p.notice= text("none")
- elsif @product_users.empty? && @search_action_performed
  - # search results returned empty
  %p.notice= text("no_results")
-  elsif @product_users.empty?
  - # no users have access to this product, no search was performed
  %p.notice= text("empty_access_list")
```

Reply:
When the product doesn't require approval, it just never loads this partial (tab isn't available). Maybe `@product_users.nil?` is unnecessary after all?

Reply:
Hm, I don't think we will hit `@product_users.empty?` in this context only when searching, right?  My instinct is to pull more of the `index` template (starting at line 23) into this partial so we can define the logic in one place.  I think it will be easier to reason about that way.  What do you think?